### ISSUE TITLE
Separate merge batches for some paths in the repository

### ISSUE THREAD

**Issue by qaifshaikh** [kubernetes/test-infra#24161]
**What would you like to be added**:
Would be nice to have a way to separate merge batches for some PRs that will have jobs that take over an hour, so that they don't slow down the smaller PRs with quicker jobs.

**Why is this needed**:
To speed up the batch merges and unblock small PRs with minor changes.
Sometimes a few jobs that take too long to complete, end up slowing down the whole pipeline because they run those tests for all PRs in the merge batches. This blocks the merge pool for small PRs for hours due to retesting the batches for each merge.
